### ISSUE TITLE
JWT addon breaks when secret contains `=`

### ISSUE THREAD

**Issue by gkalpak** [travis-ci/travis-ci#7417]
Currently, when using a secret that contains an equals sign (`=`), the JWT addon breaks. It is because when extracting the secret in order to create the JWT string ([here][1]), the `<KEY>=<secret>` string is split on all `=`, which means that if `<secret>` contains an `=`, only the part before the `=` will be used for signing the JWT.

The fix should be simply passing a second argument to `split`:

```diff
-key, secret = secret.split('=').map(&:strip)
+key, secret = secret.split('=', 2).map(&:strip)
```

Note that this is done correctly when verifying the secret [there][2].

[1]: https://github.com/travis-ci/travis-build/blob/ef82e82457f82c6518d090544bb533dc0faf0063/lib/travis/build/addons/jwt.rb#L23
[2]: https://github.com/travis-ci/travis-scheduler/blob/f50fa6373c0af5fb0da90000eb296ad7a8be3224/lib/travis/scheduler/serialize/worker/config/addons.rb#L55

**Comment by BanzaiMan** [comment]
Does Sauce Labs give you the secret containing `=`?

**Comment by gkalpak** [comment]
No. I don't know if SauceLabs give passwords containing `=`s (although that is not relevant to this issue afaict).

**Comment by BanzaiMan** [comment]
Why is it not relevant? The `jwt` addon, as currently implemented, is for for use exclusively by Sauce Connect.

**Comment by gkalpak** [comment]
Again this is just my point of view, but having a pretty useful feature such as this restricted to Sauce Connect does not make sense. The restriction is artificial and can be overcome easily. The only actual restriction is that you can only use the JWT addon for **one** service (be it Sauce Connect or anything else compatible).

That said, even this feature will remain restricted to specific services only (a.k.a  to specific variable names only :wink:), it is not unlikely to add more services in the future (hopefully), which might or might not use `=` in passwords. I can't really think of any reason you might consider keeping a bug in the code, just because it happens to work by chance (especially since it is so easy to fix).

<sub>Just m2c. I hope I'm not sounding to grumpy about it :grin:</sub>
